File: code-examples/java-rs/usage-examples/findOne.java
import com.mongodb.ConnectionString;
import com.mongodb.reactivestreams.client.MongoClient;
import com.mongodb.reactivestreams.client.MongoClients;
import com.mongodb.reactivestreams.client.MongoCollection;
import com.mongodb.reactivestreams.client.MongoDatabase;
import org.bson.Document;
import org.bson.conversions.Bson;
import java.util.Date;
import static com.mongodb.client.model.Filters.eq;
import static com.mongodb.client.model.Projections.excludeId;
import static com.mongodb.client.model.Projections.fields;
import static com.mongodb.client.model.Projections.include;
import static com.mongodb.client.model.Sorts.descending;

class IMDb {
    int rating;
    int votes;
    int id;
}

class Movie {
    String title;
    int year;
    Date released;
    String plot;
    String type;
    IMDb imdb;
}

class MovieSummary {
    String title;
    IMDb imdb;
}

public class Example {
    private static final String URI = "<connection string uri>";
    private static MongoClient mongoClient = null;
    private static MongoCollection<Document> movies = null;

    static {
        mongoClient = MongoClients.create(new ConnectionString(URI));
        MongoDatabase database = mongoClient.getDatabase("sample_mflix");
        movies = database.getCollection("movies");
    }

    public static void main(String[] args) {
        try {
            Bson filter = eq("title", "The Room");
            Bson projection = fields(include("title", "imdb"), excludeId());
            Bson sort = descending("imdb.rating");

            movies
                .find(filter)
                .projection(projection)
                .sort(sort)
                .first()
                .subscribe(movie -> System.out.println(movie.toJson()));
        } finally {
            if(mongoClient != null) {
                mongoClient.close();
            }
        }
    }
}

File: code-examples/scala/usage-examples/findOne.scala
import org.mongodb.scala._

case class IMDB(rating: Int, votes: Int, id: Int)

trait Movie {
  val title: String
  val year: Int
  val released: java.util.Date
  val plot: String
  val `type`: String
  val imdb: IMDB
}

type MovieSummary = (String, IMDB)

val uri = "<connection string uri>"
val mongoClient: MongoClient = MongoClient(uri)
val database = mongoClient.getDatabase("sample_mflix")
val collection = database.getCollection("movies")

def findMovie(): Unit = {
  val movie = collection.find(equal("title", "The Room"))
    .sort(descending("rating"))
    .projection(fields(include("title", "imdb"), excludeId()))
    .first()
    
  movie.subscribe(new Observer[Document] {
    override def onNext(result: Document): Unit = println(result)
    override def onError(e: Throwable): Unit = println(s"Failed: $e")
    override def onComplete(): Unit = println("Completed")
  })
}

findMovie()

mongoClient.close()
